### PR TITLE
[CIVisibility] - Correctly extract commit message from AppVeyor

### PR DESCRIPTION
**Pull request by albertvaka**
## Summary of changes
Fix getting the full commit message when running on the AppVeyor CI.

## Reason for change
Previously we were only checking `APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED` which, [as per the docs](https://www.appveyor.com/docs/environment-variables/) only contains the "rest" of the commit message. The first line of the commit message is in `APPVEYOR_REPO_COMMIT_MESSAGE`.

## Test coverage
Updated appveyor fixture.

### CODE REVIEW

File: tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs [hunk 1]
```diff
@@ -567,7 +567,12 @@ private void SetupAppveyorEnvironment()
                 Branch = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_BRANCH");
             }
 
-            Message = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED");
+            Message = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE");
+            string extendedMessage = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED");
+            if !string.IsNullOrWhiteSpace(extendedMessage)
+            {
+                Message = Message + "\n" + extendedMessage
+            }
```
Comment (tonyredondo) [inline tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs:575]:
```suggestion
            if (!string.IsNullOrWhiteSpace(extendedMessage))
            {
                Message = Message + "\n" + extendedMessage
            }
```

File: tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs [hunk 2]
```diff
@@ -567,7 +567,12 @@ private void SetupAppveyorEnvironment()
                 Branch = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_BRANCH");
             }
 
-            Message = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED");
+            Message = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE");
+            string extendedMessage = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED");
+            if !string.IsNullOrWhiteSpace(extendedMessage)
+            {
+                Message = Message + "\n" + extendedMessage
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs:574]: Given the "standard" format of log messages, does this need to have a double line break?
```suggestion
                Message = Message + "\n\n" + extendedMessage
```

**Review by andrewlock — APPROVED** [review]
Thanks!

File: tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs (hunk 2 above)
Comment (albertvaka) [reply tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs]: `APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED` contains everything else after the first line, so it should actually start with a newline in most cases.

**Review by tonyredondo — APPROVED** [review]
LGTM

File: tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs [hunk 3]
```diff
@@ -600,7 +600,12 @@ private void SetupAppveyorEnvironment()
...
             }
 
-            Message = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED");
+            Message = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE");
+            string extendedMessage = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_MESSAGE_EXTENDED");
+            if (!string.IsNullOrWhiteSpace(extendedMessage))
+            {
+                Message = Message + "\n" + extendedMessage;
+            }
             AuthorName = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_AUTHOR");
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs:609]: Causing build failures due to linting:
```suggestion
            }

            AuthorName = EnvironmentHelpers.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT_AUTHOR");
```

**Comment by andrewlock** [comment]
## Benchmarks Report :snail:

Benchmarks for #3183 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3): 
* 1 benchmarks are **slower**, with geometric mean 1.134
* All benchmarks have **the same** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`WriteAndFlushEnrichedTraces`|net472|720μs|916ns|3.55μs|0.359| 0|0|3.18&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|458μs|649ns|2.52μs|0| 0|0|2.58&nbsp;KB|
|#3183|`WriteAndFlushEnrichedTraces`|net472|712μs|712ns|2.66μs|0.359| 0|0|3.18&nbsp;KB|
|#3183|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|474μs|390ns|1.51μs|0| 0|0|2.59&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AppSecBodyBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`AllCycleSimpleBody`|net472|190ns|0.13ns|0.467ns|0.0676| 0|0|425&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`AllCycleSimpleBody`|netcoreapp3.1|236ns|0.322ns|1.25ns|0.00581| 0|0|424&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`AllCycleMoreComplexBody`|net472|188ns|0.105ns|0.377ns|0.0638| 0|0|401&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`AllCycleMoreComplexBody`|netcoreapp3.1|236ns|0.297ns|1.11ns|0.00546| 0|0|400&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`BodyExtractorSimpleBody`|net472|256ns|0.244ns|0.946ns|0.0573| 0|0|361&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`BodyExtractorSimpleBody`|netcoreapp3.1|222ns|0.191ns|0.714ns|0.00369| 0|0|272&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`BodyExtractorMoreComplexBody`|net472|14.4μs|17.1ns|66.1ns|1.21| 0.0217|0|7.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`BodyExtractorMoreComplexBody`|netcoreapp3.1|12.4μs|17.2ns|64.2ns|0.0869| 0|0|6.75&nbsp;KB|
|#3183|`AllCycleSimpleBody`|net472|186ns|0.15ns|0.562ns|0.0676| 0|0|425&nbsp;B|
|#3183|`AllCycleSimpleBody`|netcoreapp3.1|237ns|0.25ns|0.969ns|0.00576| 0|0|424&nbsp;B|
|#3183|`AllCycleMoreComplexBody`|net472|181ns|0.193ns|0.748ns|0.0637| 0|0|401&nbsp;B|
|#3183|`AllCycleMoreComplexBody`|netcoreapp3.1|237ns|0.176ns|0.68ns|0.00545| 0|0|400&nbsp;B|
|#3183|`BodyExtractorSimpleBody`|net472|256ns|0.241ns|0.903ns|0.0574| 0|0|361&nbsp;B|
|#3183|`BodyExtractorSimpleBody`|netcoreapp3.1|222ns|0.147ns|0.57ns|0.00369| 0|0|272&nbsp;B|
|#3183|`BodyExtractorMoreComplexBody`|net472|15.1μs|17.6ns|68ns|1.21| 0.0152|0|7.62&nbsp;KB|
|#3183|`BodyExtractorMoreComplexBody`|netcoreapp3.1|12.3μs|8.94ns|33.4ns|0.0918| 0|0|6.75&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`SendRequest`|netcoreapp3.1|181μs|316ns|1.23μs|0.271| 0|0|20.6&nbsp;KB|
|#3183|`SendRequest`|net472|0ns|0ns|0ns|0| 0|0|0&nbsp;b|
|#3183|`SendRequest`|netcoreapp3.1|183μs|202ns|781ns|0.183| 0|0|20.58&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Slower :warning: Same allocations :heavy_check_mark:



#### Slower :warning: in #3183

|  Benchmark         | diff/base | Base Median (ns) | Diff Median (ns) | Modality   |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.DbCommandBenchmark.ExecuteNonQuery&#8209;net472 | 1.134 | 1,730.43 | 1,961.72 | 

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`ExecuteNonQuery`|net472|1.73μs|2.1ns|7.84ns|0.15| 0|0|947&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`ExecuteNonQuery`|netcoreapp3.1|1.43μs|0.867ns|3.24ns|0.0122| 0|0|936&nbsp;B|
|#3183|`ExecuteNonQuery`|net472|1.96μs|2.69ns|10.4ns|0.15| 0.000982|0|947&nbsp;B|
|#3183|`ExecuteNonQuery`|netcoreapp3.1|1.47μs|1.09ns|4.09ns|0.0124| 0|0|936&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`CallElasticsearch`|net472|2.52μs|2.8ns|10.8ns|0.183| 0|0|1.16&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`CallElasticsearch`|netcoreapp3.1|1.57μs|0.913ns|3.42ns|0.0149| 0|0|1.1&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`CallElasticsearchAsync`|net472|2.73μs|1.81ns|7.01ns|0.204| 0|0|1.29&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`CallElasticsearchAsync`|netcoreapp3.1|1.61μs|0.714ns|2.67ns|0.0162| 0|0|1.22&nbsp;KB|
|#3183|`CallElasticsearch`|net472|2.58μs|1.9ns|7.37ns|0.183| 0|0|1.16&nbsp;KB|
|#3183|`CallElasticsearch`|netcoreapp3.1|1.64μs|0.761ns|2.85ns|0.0146| 0|0|1.1&nbsp;KB|
|#3183|`CallElasticsearchAsync`|net472|2.74μs|2.21ns|8.26ns|0.205| 0|0|1.29&nbsp;KB|
|#3183|`CallElasticsearchAsync`|netcoreapp3.1|1.6μs|0.757ns|2.83ns|0.0159| 0|0|1.22&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`ExecuteAsync`|net472|2.72μs|7.54ns|28.2ns|0.225| 0.00135|0|1.42&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`ExecuteAsync`|netcoreapp3.1|1.76μs|3.48ns|13.5ns|0.0185| 0|0|1.34&nbsp;KB|
|#3183|`ExecuteAsync`|net472|2.76μs|5.69ns|21.3ns|0.226| 0|0|1.42&nbsp;KB|
|#3183|`ExecuteAsync`|netcoreapp3.1|1.67μs|5.26ns|20.4ns|0.0183| 0|0|1.34&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`SendAsync`|net472|5.67μs|8.76ns|33.9ns|0.44| 0|0|2.77&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`SendAsync`|netcoreapp3.1|3.64μs|7.35ns|27.5ns|0.0344| 0.00181|0|2.6&nbsp;KB|
|#3183|`SendAsync`|net472|5.78μs|7.9ns|30.6ns|0.439| 0|0|2.77&nbsp;KB|
|#3183|`SendAsync`|netcoreapp3.1|3.71μs|11.7ns|45.3ns|0.0343| 0|0|2.6&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`EnrichedLog`|net472|3.22μs|1.67ns|6.49ns|0.287| 0|0|1.81&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`EnrichedLog`|netcoreapp3.1|2.59μs|0.974ns|3.51ns|0.0254| 0|0|1.85&nbsp;KB|
|#3183|`EnrichedLog`|net472|3.22μs|7ns|27.1ns|0.288| 0|0|1.81&nbsp;KB|
|#3183|`EnrichedLog`|netcoreapp3.1|2.49μs|1.15ns|4.29ns|0.0249| 0|0|1.85&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`EnrichedLog`|net472|153μs|216ns|835ns|0.684| 0.076|0|4.66&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`EnrichedLog`|netcoreapp3.1|118μs|304ns|1.18μs|0.0586| 0|0|4.49&nbsp;KB|
|#3183|`EnrichedLog`|net472|156μs|203ns|788ns|0.721| 0.24|0|4.66&nbsp;KB|
|#3183|`EnrichedLog`|netcoreapp3.1|118μs|282ns|1.09μs|0.0582| 0|0|4.49&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`EnrichedLog`|net472|5.59μs|10.3ns|37.1ns|0.569| 0.0028|0|3.59&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`EnrichedLog`|netcoreapp3.1|4.28μs|7.24ns|27.1ns|0.0525| 0.0021|0|3.91&nbsp;KB|
|#3183|`EnrichedLog`|net472|5.72μs|15.5ns|60.2ns|0.569| 0.00283|0|3.59&nbsp;KB|
|#3183|`EnrichedLog`|netcoreapp3.1|4.22μs|8.72ns|33.8ns|0.0529| 0|0|3.91&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`SendReceive`|net472|2.37μs|1.15ns|4.29ns|0.218| 0|0|1.37&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`SendReceive`|netcoreapp3.1|1.88μs|1.97ns|7.36ns|0.0181| 0|0|1.32&nbsp;KB|
|#3183|`SendReceive`|net472|2.4μs|1.88ns|7.27ns|0.218| 0|0|1.37&nbsp;KB|
|#3183|`SendReceive`|netcoreapp3.1|1.8μs|0.946ns|3.41ns|0.0181| 0|0|1.32&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`EnrichedLog`|net472|5.06μs|3.59ns|13.4ns|0.354| 0.00253|0|2.23&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`EnrichedLog`|netcoreapp3.1|4.36μs|1.32ns|5.12ns|0.0239| 0|0|1.8&nbsp;KB|
|#3183|`EnrichedLog`|net472|5.06μs|2.09ns|7.8ns|0.353| 0|0|2.23&nbsp;KB|
|#3183|`EnrichedLog`|netcoreapp3.1|4.14μs|1.61ns|6.02ns|0.0227| 0|0|1.8&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`StartFinishSpan`|net472|1.17μs|0.483ns|1.81ns|0.129| 0|0|810&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`StartFinishSpan`|netcoreapp3.1|956ns|0.403ns|1.45ns|0.00996| 0|0|760&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`StartFinishScope`|net472|1.41μs|0.587ns|2.27ns|0.142| 0|0|891&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`StartFinishScope`|netcoreapp3.1|1.07μs|0.637ns|2.3ns|0.0119| 0|0|880&nbsp;B|
|#3183|`StartFinishSpan`|net472|1.15μs|0.67ns|2.59ns|0.128| 0|0|810&nbsp;B|
|#3183|`StartFinishSpan`|netcoreapp3.1|997ns|0.382ns|1.38ns|0.0105| 0|0|760&nbsp;B|
|#3183|`StartFinishScope`|net472|1.36μs|1.71ns|6.62ns|0.141| 0|0|891&nbsp;B|
|#3183|`StartFinishScope`|netcoreapp3.1|1.11μs|1.72ns|6.65ns|0.0118| 0|0|880&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`RunOnMethodBegin`|net472|1.5μs|1.05ns|4.07ns|0.142| 0|0|891&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/55542008e969d25a704349de09eada3a71e5c8c3)|`RunOnMethodBegin`|netcoreapp3.1|1.13μs|0.639ns|2.39ns|0.0119| 0|0|880&nbsp;B|
|#3183|`RunOnMethodBegin`|net472|1.53μs|1.17ns|4.54ns|0.141| 0|0|891&nbsp;B|
|#3183|`RunOnMethodBegin`|netcoreapp3.1|1.18μs|0.557ns|2.08ns|0.0118| 0|0|880&nbsp;B|
</details>

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #3183 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/65c6c4e95b8092d1e4608d482f5e91a8044eaf42) will **not change** line coverage
:heavy_check_mark: Merging #3183 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/65c6c4e95b8092d1e4608d482f5e91a8044eaf42) will **not change** branch coverage
:no_entry: Merging #3183 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/65c6c4e95b8092d1e4608d482f5e91a8044eaf42) will will **increase** complexity by `21`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/65c6c4e95b8092d1e4608d482f5e91a8044eaf42) | #3183       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `17971` / `24517` | `18033` / `24569` |          |
| Lines %   | `73%`      | `73%`      |  `0%` :heavy_check_mark:  |
| Branches  | `10585` / `15050` | `10612` / `15058` |          |
| Branches %| `70%`      | `70%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `16261`      | `16282`        |  `21`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=111938&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/111938/artifacts?artifactName=Code%20Coverage%20Report_111938&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3183](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=112141&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/112141/artifacts?artifactName=Code%20Coverage%20Report_112141&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/65c6c4e95b8092d1e4608d482f5e91a8044eaf42) | #3183       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `73%`      | `73%`       |  `0%` :heavy_check_mark:  |
| Branches %| `70%`      | `70%`       |  `0%` :heavy_check_mark:  |
| Complexity| `16261`      | `16282`       |  `21` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.RemoteConfigurationManagement.RemoteConfigurationCache](https://github.com/DataDog/dd-trace-dotnet/tree/e93f893adec76c5bf59246886dd37118d0faef54/tracer/src/Datadog.Trace/RemoteConfigurationManagement/RemoteConfigurationCache.cs) | `-86%` :no_entry: | `0%` :heavy_check_mark:   | `2` :no_entry: |
| [Datadog.Trace.RemoteConfigurationManagement.ProductConfigChangedEventArgs](https://github.com/DataDog/dd-trace-dotnet/tree/e93f893adec76c5bf59246886dd37118d0faef54/tracer/src/Datadog.Trace/RemoteConfigurationManagement/ProductConfigChangedEventArgs.cs) | `-68%` :no_entry: | `-67%` :no_entry:   | `6` :no_entry: |
| [Datadog.Trace.RemoteConfigurationManagement.Product](https://github.com/DataDog/dd-trace-dotnet/tree/e93f893adec76c5bf59246886dd37118d0faef54/tracer/src/Datadog.Trace/RemoteConfigurationManagement/Product.cs) | `-31%` :no_entry: | `-29%` :no_entry:   | `9` :no_entry: |
| [Datadog.Trace.Ci.CIVisibility](https://github.com/DataDog/dd-trace-dotnet/tree/e93f893adec76c5bf59246886dd37118d0faef54/tracer/src/Datadog.Trace/Ci/CIVisibility.cs) | `5%` :heavy_check_mark: | `6%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/e93f893adec76c5bf59246886dd37118d0faef54/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `5%` :heavy_check_mark: | `8%` :heavy_check_mark:   | `0` :heavy_check_mark: |


The following classes were added in #3183:

| File    | Line coverage | Branch coverage | Complexity |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.RemoteConfigurationManagement.ApplyDetails](https://github.com/DataDog/dd-trace-dotnet/pull/3183/files) | `0%` | `100%` | `2` |
| [Datadog.Trace.RemoteConfigurationManagement.NamedRawFile](https://github.com/DataDog/dd-trace-dotnet/pull/3183/files) | `100%` | `100%` | `1` |
| [Datadog.Trace.RemoteConfigurationManagement.NamedTypedFile\`1](https://github.com/DataDog/dd-trace-dotnet/pull/3183/files) | `100%` | `100%` | `1` |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=111938&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/111938/artifacts?artifactName=Code%20Coverage%20Report_111938&api-version=6.0&%24format=zip)  
* [HTML report for this PR #3183](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=112141&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/112141/artifacts?artifactName=Code%20Coverage%20Report_112141&api-version=6.0&%24format=zip)

**Pull request merged**